care/emr/services/scheduling_service.py

```
@@ -0,0 +1,24 @@
+    def get_available_slots(facility, request_data):
+        """
+        Pure business logic: Calculates slots without knowing about HTTP requests.
+        """
+        # In the real code, we would call the helper function here
+        resource = get_schedulable_resource(
+            request_data.resource_type,
+            request_data.resource_id,
+            facility,
+        )
+        if not resource:
+            raise ValidationError("No schedules found for this resource")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)